app/models/portfolio.rb

```diff
@@ -22,7 +24,25 @@ def add_portfolio_item(portfolio_item)
   end
 
   def metadata
-    {:user_capabilities => user_capabilities,
-     :shared            => self.access_control_entries.any?}
+    {:user_capabilities => user_capabilities}.tap do |data|
+      data[:statistics] = statistics
+    end
+  end
+
+  def update_statistics
+    current_stats = stats
+    return if current_stats.eql?(statistics)
+
+    update(:statistics => current_stats)
+  end
+
+  private
+
+  def stats
```

Comment:
@gmcculloug should this stats be present in the PortfolioItem too since they can also have approvals on them and maybe other stats down the road, e.g. number of times the product has been installed.

Reply:
@gmcculloug if we move this to a mixin we could reuse this for Portfolio and PortfolioItems.
Maybe here
https://github.com/RedHatInsights/catalog-api/blob/master/app/models/concerns/user_capabilities.rb
or a new mixin